Zad1 - front-end

karienne opened the pull request:
@lukasz-galka @alexsimkovich

karienne committed:
Add files via upload

karienne committed:
Delete README.md

alexsimkovich commented on main.js:
```diff
@@ -0,0 +1,203 @@
+url = 'https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json';
```
proszę poprawnie zdefiniować zmienną

alexsimkovich commented on main.js:
```diff
@@ -0,0 +1,203 @@
+
+
+function appendData(data){ //załadowanie danych na stronie
+
```
można to usprawnić, np. użyć https://developer.mozilla.org/pl/docs/Web/API/Element/insertAdjacentHTML

```
function appendData(data) {
    const column1 = document.getElementById("col1");

    data.forEach(item => {
        productsInCart.push({ 'inCart': 0 })
        column1.insertAdjacentHTML('afterbegin', `
          <div class='product'>
            <img src='${item.image}' class='product-image' />
            <h3 class='title-product'>${item.title}</h3>
            <h3 class='price-product'>${item.price.toFixed(2)}</h3>
            <h3 class='ingredients-product'>${item.ingredients.join(', ')}</h3>
            <a class='add-cart cart-button'>Zamów</a>
          </div>
    `)
    })
}
```

alexsimkovich commented on main.js:
```diff
@@ -0,0 +1,203 @@
+
+function appendData(data){ //załadowanie danych na stronie
+
+    let column1 = document.getElementById("col1");
```
to jest `const`

alexsimkovich commented on main.js:
```diff
@@ -0,0 +1,203 @@
+}
+
+function checkCart(){ //sprawdzanie czy koszyk jest pusty
+    let cart = document.getElementById("col2");
```
proszę przeczytać czym się różnią let od const. Naprzykład https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

alexsimkovich commented on main.js:
```diff
@@ -0,0 +1,203 @@
+
+    if (productsInCart[index].inCart == 0){
+
+        let sum = document.getElementById("tpn");
```
Też bym sugerował pójść w stronę https://developer.mozilla.org/pl/docs/Web/API/Element/insertAdjacentHTML

lukasz-galka commented on index.html:
```diff
@@ -0,0 +1,41 @@
+    <div class="container">
+        <div class="column1">
+            <h2 class="menu-text">Menu</h2>
+            <div class="products-list" id="col1">
```
`col1` i `col2` nie sa dobrymi nazwami dla id, nadaj bardziej opisowe id

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+    if(inCartItems != 0 && emptyCart){
+        cart.removeChild(emptyCart);
+    }
+    else if(inCartItems == 0 && !emptyCart){
```
emptyCard moze byc caly czas w htmlu, nadaj na rodzica klase ktora mowi czy koszyk jest pusty i na jej podstawie schowaj/pokaz ten napis

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+function setItemInCart(product){ //umieszczenie produktu w koszyku
+    let cart = document.getElementById("col2");
+    let index = product.id - 1;
+    let total = document.getElementById("tp"); //koszt zamówienia
```
`col2`, `tp` - nadaj bardziej opisowe `id`

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+
+function setItemInCart(product){ //umieszczenie produktu w koszyku
+    let cart = document.getElementById("col2");
+    let index = product.id - 1;
```
nie mozesz zalozyć ze `id` ktore dostajesz w api zawsze bedą po kolei, z czasem jakas pizza moze byc skasowana i wtedy dostaniesz id np `1 2 4...`

karienne replied:
dziękuję za sugestię :) czy jest zatem jakiś sposób bym mogła dodać kolejny parametr (inCart) do każdego produktu w bazie, tak bym nie musiała tworzyć dodatkowej tablicy przechowującej informacje czy produkt jest w koszyku i w jakiej ilości?

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+        cart.appendChild(total);
+    }
+
+    if (productsInCart[index].inCart == 0){
```
skorzystaj z `===`

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+        ingredients.innerHTML = data[i].ingredients.join(", ");
+        div.appendChild(ingredients);
+
+        button = document.createElement("a");
```
to `button` czy `link` ?

lukasz-galka commented on main.js:
```diff
@@ -0,0 +1,203 @@
+
+        let totalPriceTitle = document.createElement("h4");
+        totalPriceTitle.classList.add('total-price-title');
+        totalPriceTitle.innerHTML = "Koszt zamówienia: ";
```
takie elementy powinny byc od razy w htmlu i tylko powinnas je chowac/pokazywac jak koszyk jest pusty lub nie poprzez dodanie odpowiedniej klasy

karienne committed:
Update index.html

Poprawa nazw id

karienne committed:
Update main.js

1. Poprawa deklaracji zmiennych (const i let)
2. Zmiana sposobu liczenia produktów w koszyku - zastąpienie dodatkowej tablicy parametrem "counter" dla każdego produktu w pliku "data"
3. Użycie funkcji insertAdjacentHTML() zamiast ręcznego dodawania elementów HTML (createElement(), appendChild())
4. Dodanie klasy mówiącej, czy koszyk jest pusty czy pełny i na ich podstawie pokazywanie/chowanie napisu w koszyku

karienne committed:
Update style.css

Dodanie klasy 'full-cart', która jest odpowiedzialna za ukrywanie napisu w koszyku, kiedy jest on pełny
